source/MaterialXView/ShaderValidators/Osl/OslValidator.cpp

```diff
@@ -36,12 +37,16 @@ void OslValidator::compileOSL(const std::string& oslFileName)
         return;
     }
 
+    // Remove .osl and add .oso extension for output. 
+    std::string outputFileName = removeExtension(oslFileName);
+    outputFileName += ".oso";
+
     // Use a known error file name to check
     std::string errorFile(oslFileName + "_errors.txt");
     const std::string redirectString(" 2>&1");
 
     // Run the command and get back the result. If non-empty string throw exception with error
-    std::string command = _oslCompilerExecutable + " -q -I\"" + _oslIncludePathString + "\" " + oslFileName + " > " +
+    std::string command = _oslCompilerExecutable + " -q -I\"" + _oslIncludePathString + "\" " + oslFileName + " -o " + outputFileName + " > " +
```

Comment:
Add in output file name option which we want to be the root osl file name but .osl replaced with .oso. testshade requires that the filename match the shader name to test with. if the osl file is called foo.osl, the shader name to test is called foo, and foo.oso is searched for.